test/e2e_test.go

```diff
@@ -0,0 +1,166 @@
+// +build e2e
+
+package test
+
+import (
+	"context"
+	"github.com/Shopify/sarama"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+)
+
+const (
+	httpUrlPrefix   = "http://localhost:8080"
+	metricsEndpoint = "/metrics"
+)
+
+/* test checks for following:
+*  the presence of canary topic,
+*  liveliness of topic (messages being produced),
+*/
+func TestCanaryTopicLiveliness(t *testing.T) {
+	log.Println("TestCanaryTopic test starts")
+
+	// setting up timeout and handler for consumer group, there is no need to wait much longer then is the retention time.
+	timeout := time.After(time.Second * 10)
+	handler := exampleConsumerGroupHandler{}
+	handler.mutexWritePartitionPresence = &sync.Mutex{}
+	handler.consumingDone = make(chan bool)
+
+	// test itself.
+	go func() {
+		config := sarama.NewConfig()
+		config.Consumer.Return.Errors = true
+		config.Consumer.Offsets.Initial =  sarama.OffsetOldest
+		ctx := context.Background()
+
+		//kafka end point
+		brokers := []string{serviceManager.KafkaBrokerAddress}
+		//get broker
+		consumer, err := sarama.NewConsumer(brokers, config)
+
+		if err != nil {
+			t.Fatalf(err.Error())
+		}
+		// get all topics
+		topics, err := consumer.Topics()
+		if err != nil {
+			t.Fatalf(err.Error())
+		}
+		if !IsTopicPresent(serviceManager.TopicTestName, topics) {
+			t.Fatalf("%s is not present amongst existing topic %v", serviceManager.TopicTestName, topics)
+		}
+		// consume single message
+		group, err := sarama.NewConsumerGroup([]string{serviceManager.KafkaBrokerAddress}, "faq-g9", config)
+		if err != nil {
+			panic(err)
+		}
+
+		// set up client for getting partition count on canary topic
+		client, _ := sarama.NewClient([]string{serviceManager.KafkaBrokerAddress},config )
+		listOfPartitions, _ := client.Partitions(serviceManager.TopicTestName)
+		var partitionsCount =  len(listOfPartitions)
+		handler.partitionsConsumptionSlice = make([]bool, partitionsCount)
```

Comment:
I think that for getting topic existence and partitions you should use the same approach as the topic.go file. It's using the admin client to describe topics, getting just the one you need to check, without then looping in a collection and getting metadata with partitions info as well.